test/cppsuite/tests/hs_cleanup.cxx

```diff
@@ -93,11 +95,16 @@ class hs_cleanup : public test {
              */
             if (tc->update(cursor, coll.id, key_value_t(key_tmp))) {
                 if (tc->transaction.can_commit()) {
-                    tc->transaction.commit();
+                    if (tc->transaction.commit())
+                        rollback_retries = 0;
+                    else
+                        ++rollback_retries;
                 }
             } else {
                 tc->transaction.rollback();
+                ++rollback_retries;
             }
+            testutil_assert(rollback_retries < MAX_ROLLBACKS);
         }
         /* Ensure our last transaction is resolved. */
         if (tc->transaction.active())
```

Comment:
Should we not increment `rollback_retries` here as well and then assert rollback_retries < MAX_ROLLBACKS?